### ISSUE TITLE
[MNG-755] Remove the dysfunctional command line arguments

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3753]
**[Bob Allison](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=allisonb)** opened **[MNG-755](https://issues.apache.org/jira/browse/MNG-755?redirect=false)** and commented

I ran "m2 -fae clean:clean package site:site" on my top level package to clean up all targets, rebuild the artifacts, and generate a site for every project (I used -fae since I have a problem building the site in one sub-project).  The banner displayed when it started processing was:
[INFO] ----------------------------------------------------------------------------
[INFO] Building Maven Default Project
[INFO]    task-segment: [clean:clean, package, site:site]
[INFO] ----------------------------------------------------------------------------
even though I have a different name in the POM.

A little later, the build fails with:
[INFO] [jar:jar]
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] ----------------------------------------------------------------------------
[ERROR] BUILD ERROR
[INFO] ----------------------------------------------------------------------------
[INFO] Diagnosis: Error assembling JAR
[INFO] ----------------------------------------------------------------------------
[ERROR] Cause:
org.apache.maven.plugin.MojoExecutionException: Error assembling JAR
at org.apache.maven.plugin.jar.JarMojo.execute(JarMojo.java:107)
at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:364)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:470)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalWithLifecycle(DefaultLifecycleExecutor.java:443)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:429)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:266)
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:125)
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:186)
at org.apache.maven.cli.MavenCli.main(MavenCli.java:316)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
at java.lang.reflect.Method.invoke(Method.java:585)
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: java.lang.NullPointerException
at org.apache.maven.archiver.MavenArchiver.createArchive(MavenArchiver.java:238)
at org.apache.maven.plugin.jar.JarMojo.execute(JarMojo.java:100)
... 16 more
which is caused by not having any sources in the project (since it is a POM project).  The build also did not run any of the modules.

---

**Affects:** 2.0-beta-1

**Issue Links:**
- [MNG-720](https://issues.apache.org/jira/browse/MNG-720) short commandline options are not parsed
 (_**"supercedes"**_)

**Comment by jira-importer** [comment]
**[Bob Allison](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=allisonb)** commented

It looks like "m2 -fn" exhibits the same behavior.

**Comment by jira-importer** [comment]
**[Bob Allison](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=allisonb)** commented

I wasn't sure if this was one bug or two.  I did not know if the NPE was a result of using the "Default Maven Project" or something different.

Using --fail-at-end instead of -fae causes the build to use the POM in the directory rather than a default POM.  I also do not get the NPE.  I guess this can be closed since the whole thing was caused by using the documented but disfunctional command line options.
